operators/src/pro/ml/xgboost.rs

```diff
@@ -0,0 +1,628 @@
+        let vec_of_rqps: Vec<Box<dyn RasterQueryProcessor<RasterType = f32>>> = self
+            .sources
+            .iter()
+            .map(|init_raster| init_raster.query_processor().unwrap().into_f32())
+            .collect();
+
+        Ok(QueryProcessorOut(Box::new(XgboostProcessor::new(
+            vec_of_rqps,
+            self.model_file_path.clone(),
+            self.no_data_value,
+        ))))
+    }
+}
+
+struct XgboostProcessor<Q, P>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+{
+    sources: Vec<Q>,
+    model_file: PathBuf,
+    no_data_value: Option<f32>,
+}
+
+impl<Q, P> XgboostProcessor<Q, P>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+    P: Pixel,
+{
+    pub fn new(sources: Vec<Q>, model_file_path: PathBuf, no_data_value: Option<f32>) -> Self {
+        Self {
+            sources,
+            model_file: model_file_path,
+            no_data_value,
+        }
+    }
+
+    async fn process_tile_async(
+        &self,
+        mut bands_of_tile: Vec<Result<RasterTile2D<P>, crate::error::Error>>,
+        pool: Arc<ThreadPool>,
+    ) -> Result<RasterTile2D<PixelOut>> {
+        let tile = bands_of_tile.pop().ok_or(error::Error::EmptyInput)??;
+
+        // gather the data
+        let grid_shape = tile.grid_shape();
+        let n_bands = bands_of_tile.len() as i32 + 1; // +1 because of .pop()
+        let props = tile.properties.clone(); // = &tile.properties;
+        let time = tile.time;
+        let tile_position = tile.tile_position;
+        let global_geo_transform = tile.global_geo_transform;
+
+        let model = self.model_file.clone();
+        let ndv = self.no_data_value.or(Some(f32::NAN));
+
+        let predicted_grid: geoengine_datatypes::raster::Grid<GridShape<[usize; 2]>, f32> =
+            crate::util::spawn_blocking(move || {
+                // put the tile back into place
+                bands_of_tile.insert(0, Ok(tile));
+                // extract the actual data from the tiles
+                let pixels: Vec<_> = bands_of_tile
+                    .into_iter()
+                    .flat_map(|band| {
+                        let band_ok = match band {
+                            Ok(x) => x,
+                            Err(err) => {
+                                panic!("Failed to get BaseTile with error msg: \n{:?}", err)
+                            }
+                        };
+
+                        let mat_tile = band_ok.into_materialized_tile();
+
+                        mat_tile
+                            .grid_array
+                            .inner_grid
+                            .data
+                            .into_iter()
+                            .map(num_traits::AsPrimitive::as_)
+                            .collect::<Vec<f32>>()
+                    })
+                    .collect();
+
+                // let pixels: Vec<_> = rasters.into_iter().collect();
+                let model_content = std::fs::read_to_string(model).unwrap();
```

Comment:
you should never use unwrap, but always handle the error. Or: if you are sure that it can never occure, use `expect` and explain why it can never fail.

Here it CAN fail and it should result in an Error and not a crash. How large is the model? Depending on the size I would suggest to load the model during `initialize` of the Operator and just keep it in memory. We don't want to read it for every tile. 

If it can be very large then maybe only load it on `query` because if we load it during `initialize` we may have loaded in unnecessarily when the processor is never actually queried.

Reply:
I changed the error handling with https://github.com/geo-engine/geoengine/commit/84fb951c583cb7141ff50f1196f90504d2ef8a90.

There is an ugly problem with the model file. I need to load the model for each tile, because the booster instance has to be created distinctly and can not be used across the multiple tiles. I encountered this issue a while ago and am not entirely sure anymore, but I think it had to do with the unsafe part of the booster code. 

I will try to recreate the situation, maybe we can sort it out then.

Reply:
ok but you can still read the file once and only recreate the booster instance for each tile, right?

Reply:
https://github.com/geo-engine/geoengine/pull/639/commits/4d0ada181995c7c47fdc471a097241750233ffc1